extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -358,8 +390,29 @@ func (a *genericActuator) waitUntilWantedMachineDeploymentsAvailable(ctx context
 			if health.CheckMachineDeployment(&deployment) == nil {
 				numHealthyDeployments++
 			}
-			numDesired += deployment.Spec.Replicas
-			numUpdated += deployment.Status.UpdatedReplicas
+
+			if gardenerutils.IsMachineDeploymentStrategyManualInPlace(deployment.Spec.Strategy) {
+				oldMachineSetsTotalReplicas := 0
```

Comment:
```suggestion
				oldMachineSetsTotalReplicas := int32(0)
```
To avoid casting in other places.